libgnucash/engine/gnc-budget.cpp

```diff
@@ -758,6 +758,8 @@ get_perioddata (const GncBudget *budget, const Account *account, guint period_nu
 {
     GncBudgetPrivate *priv = GET_PRIVATE (budget);
 
+    g_assert (period_num < priv->num_periods);
```

Comment:
Unfortunately g_assert is a no-op in a release build. Since this is a static function the best fix would be to use `return vec.at(period_num);` instead of `return vec[period_num);` and handle the `std::out_of_range` exception at each call of `get_perioddata()`.

Reply:
Or you could `thow std::out_of_range{}` yourself instead of the `assert`.